src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,43 @@
 
 class Animal {
   // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    } else {
+      this.hidden = false;
+    }
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(nameHerb) {
+    if (nameHerb.hidden === false) {
+      nameHerb.health -= 50;
+
+      if (nameHerb.health === 0) {
+        delete Animal.alive.pop(nameHerb);
+      }
+    }
+  }
```

Comment:
1) it's not a nested if condition. Both conditions should be at one level
2) update first condition 
```
    if (nameHerb.hidden === false && nameHerb instanceof Herbivore) {
      nameHerb.health -= 50;
    }
```
3) update second condition. Because what will happen if your animal has 30hp and you make a bite and take 50hp?
```
      if (nameHerb.health <= 0) {
        Animal.alive.pop(nameHerb);
      }
```

Reply:
yeah, you're right!